src/nvim/if_cscope.c

```diff
@@ -1772,11 +1777,14 @@ static int cs_read_prompt(int i)
       if (bufpos < maxlen - 1 && vim_isprintc(ch)) {
         // lazy buffer allocation
         if (buf == NULL) {
-          buf = xmalloc(maxlen);
+          assert(maxlen >= 0);
+          buf = xmalloc((size_t)maxlen);
         }
         {
           /* append character to the message */
-          buf[bufpos++] = ch;
+          assert(ch >= 0);
+          assert(ch <= CHAR_MAX);
```

Comment:
> asserts are extremely valuable, at least for the core dump they generate

I didn't mean to imply that asserts as a whole aren't meaningful (with core-dumps and whatnot), I meant that the statements printed by asserts aren't necessarily useful in themselves, as shown by the example above. 

> assert() is already a fairly ugly compromise, so tidying it up where we can seems worth it to me.

I'd argue that we should prefer practicality over style in this situation, but I'm the kind of person who thinks the comma operator is evil so I'm possible too biased :)